clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1000,8 +1000,11 @@ void invokeCompletedOffsetCommitCallbacks() {
     public RequestFuture<Void> commitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, final OffsetCommitCallback callback) {
         invokeCompletedOffsetCommitCallbacks();
 
-        RequestFuture<Void> future =  null;
-        if (!coordinatorUnknown()) {
+        RequestFuture<Void> future = null;
+        if (offsets.isEmpty()) {
+            // No need to check coordinator if offsets is empty since commit of empty offsets is completed locally.
+            future = doCommitOffsetsAsync(offsets, callback);
+        } else if (!coordinatorUnknown()) {
             future = doCommitOffsetsAsync(offsets, callback);
```

Comment:
Could we do this and add the comment above?
```java
// No need to check coordinator if offsets is empty since commit of empty offsets is completed locally.
if (offsets.isEmpty() || !coordinatorUnknown()) {
    future = doCommitOffsetsAsync(offsets, callback);
}
```

Reply:
I considered this but I think that it is clearer when kept separated.

Reply:
I'm wonder if other dev would think this is a mistake when seeing this. Or at least we should leave a comment to mention this is on purpose. WDYT?

Reply:
I checked again, and I think it's OK. Thanks.